addons/binding/org.openhab.binding.nuki/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="nuki"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+        xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+    <!-- Nuki Bridge (Bridge Type) -->
+    <bridge-type id="bridge">
+        <label>Nuki Bridge</label>
+        <description>This bridge represents a Nuki Bridge on your local network. Nuki Smart Locks have to be paired via Bluetooth with it.</description>
+        <config-description>
+            <parameter name="ip" type="text" required="true">
+                <label>IP Address</label>
+                <description>The IP Address of the Nuki Bridge.</description>
+            </parameter>
+            <parameter name="port" type="integer" required="true">
+                <label>Port</label>
+                <description>The Port on which the Nuki Bridge REST API runs.</description>
+                <default>8080</default>
+            </parameter>
+            <parameter name="apiToken" type="text" required="true">
+                <label>API Token</label>
+                <description>The API Token configured via the Nuki App when enabling the API/IFTTT.</description>
+            </parameter>
+        </config-description>
+    </bridge-type>
+
+    <!-- Nuki Smart Lock (Thing Type) -->
+    <thing-type id="smartLock">
+        <supported-bridge-type-refs>
+            <bridge-type-ref id="bridge"/>
+        </supported-bridge-type-refs>
+        <label>Nuki Smart Lock</label>
+        <description>Nuki Smart Lock which is paired via Bluetooth to a Nuki Bridge.</description>
+        <channels>
+            <channel id="smartLockOpenClose" typeId="channelTypeSmartLockOpenClose"/>
+            <channel  id="smartLockUnlatchClose" typeId="channelTypeSmartLockUnlatchClose"/>
+        </channels>
+        <config-description>
+            <parameter name="nukiId" type="text" required="true">
+                <label>Nuki ID</label>
+                <description>The ID of the Nuki Smart Lock.</description>
+            </parameter>
+        </config-description>
+    </thing-type>
+
+    <!-- Channel Type -->
+    <channel-type id="channelTypeSmartLockOpenClose">
+        <item-type>Switch</item-type>
+        <label>Nuki Smart Lock Open/Close</label>
+        <description>Nuki Smart Lock channel to open or close the door.</description>
+    </channel-type>
+
+    <channel-type id="channelTypeSmartLockUnlatchClose">
+        <item-type>Switch</item-type>
+        <label>Nuki Smart Lock Unlatch/Close</label>
+        <description>Nuki Smart Lock channel to unlatch or close the door.</description>
```

Comment:
> Oh hell, this is again a good example, why there is no way to come up with some "standard" 

:)

Thanks for the detailed information and your suggestion!

Let me summarize, just to be sure:

- add a new config-description parameter for smartlock (boolean unlatch)
- Channels
1. Channel unlock of type Switch (which uses either unlock or unlatch depending thing configuration above)
2. Channel lockAction of type advanced Number (which accepts all possible lock actions)
3. Channel lockState of type advanced read-only String to display current lock state)
4. Channel system.low-battery
5. A system.signal-strength channel

Still 5 Channels but I would really like to have 3. Channel lockState available in the binding.